chapters/en/module7.md

```diff
@@ -0,0 +1,219 @@
+</opt>
+
+<opt text="Nothing will appear" correct="true">
+Since the first cell is not executed yet, the notebook's kernel doesn't know anything about those functions.
+</opt>
+
+<opt text="You'll get an error message">
+No error error message will appear as a result of pressing <code>Shift + Tab</code>.
+</opt>
+
+<opt text="JupyterLab's contextual help window will be opened">
+The <code>Shift + Tab</code> shortcut is not set for opening the contextual help window.
+</opt>
+
+</choice>
+
+A notebook's console uses the same kernel as the notebook itself.
+
+<choice id="2">
+
+<opt text="True" correct="true">
+</opt>
+
+<opt text="False">
+Even though you can launch an independent console in JupyterLab too, the console opened for a particular notebook is connected to the same kernel as the notebook.
+</opt>
+
+</choice>
+
+We have a notebook like the following:
+
+<img src="/module7/mcq2.png" width="500"></img>
+
+What do you think the output will be if we run the last cell?
+
+<choice id="3">
+
+<opt text="27">
+Note the order in which the cells have been executed. This would have been the output if only the second cell was already run.
+</opt>
+
+<opt text="9">
+Note the order in which the cells have been executed. This would have been the output if the second and third cell were already run.
+</opt>
+
+<opt text="60" correct="true">
+</opt>
+
+<opt text="An error message">
+Note that some cells have already been executed. An error message would have resulted from running the last cell only if the kernel didn't know about the variable <code>num</code>.
+</opt>
+
+</choice>
+
+</exercise>
+<!-- ------------------------------------ -->
+<exercise id='8' title="Markdown for text formatting" type="slides,video">
+<slides source="module7/module7_04"></slides>
+</exercise>
+<!-- ------------------------------------ -->
+<exercise id='9' title="Test your knowledge">
+
+It's convenient to run code inside Markdown code blocks if you've specified the language after the first three back ticks, e.g. ` ```python`.
+
+<choice id="1">
+
+<opt text="True">
+You can't run code inside a Markdown document! Code can only be executed inside code cells in a Jupyter notebook.
+</opt>
+
+<opt text="False" correct="true">
+</opt>
+
+</choice>
+
+Markdown documents can be rendered with any text editor.
+
+<choice id="2">
+
+<opt text="True">
+You can open a Markdown document with any text editor, since it only contains plain text. However, you can view <b>rendered output</b> only in environments that support displaying Markdown output, such as JupyterLab or GitHub.
+</opt>
+
+<opt text="False" correct="true">
+</opt>
+
+</choice>
+
+</exercise>
+<!-- ------------------------------------ -->
+<exercise id='10' title="Do more with JupyterLab" type="slides,video">
+<slides source="module7/module7_05"></slides>
+</exercise>
+<!-- ------------------------------------ -->
+<exercise id='11' title="Test your knowledge">
+
+Which of the following statements is false?
```

Comment:
Could we add one more question here, maybe on one of the advanced topics I suggested?